oracle/service/runner/runner.go

```diff
@@ -65,14 +65,35 @@ func ProcessSignVoteQueue(oracleInfo *types.OracleInfo, consensusState *cs.State
 
 	// batch sign the entire unsignedVoteBuffer and add to gossipBuffer
 	newGossipVote := &oracleproto.GossipedVotes{
-		Validator:       oracleInfo.PubKey.Address(),
+		PubKey:          oracleInfo.PubKey.Bytes(),
 		SignedTimestamp: time.Now().Unix(),
 		Votes:           unsignedVotes,
 	}
 
+	// set sigPrefix based on account type and sign type
+	sigPrefix := []byte{}
+	if oracleInfo.Config.EnableSubAccountSigning {
```

Comment:
pull this encoding out? also expose decoding method so that carbon can use

Reply:
resolved in https://github.com/Switcheo/cometbft/pull/6/commits/0da46e4fc59d5032228febaf96b53a6917250976